R/names.R

```diff
@@ -509,3 +509,90 @@ vec_as_names_legacy <- function(names, prefix = "V", sep = "") {
 
   names
 }
+
+
+#' Name specifications
+#'
+#' @description
+#'
+#' A name specification describes how to combine an external name with
+#' internal names. This sort of name combination arises when
+#' concatenating vectors or flattening lists. There are two possible
+#' cases:
+#'
+#' * Named vector:
+#'
+#'   ```
+#'   vec_c(external = c(internal1 = 1, internal2 = 2))
+#'   ```
+#'
+#' * Unnamed vector:
+#'
+#'   ```
+#'   vec_c(external = 1:2)
+#'   ```
+#'
+#' In r-lib and tidyverse packages, these cases are errors by default.
```

Comment:
the cases are errors by default, because there's no behaviour that works well for every case. Instead, you can provide ...